File: DogShelter/DogShelter.Model/Volonter.cs
namespace DogShelter.Model;

public class Volonter
{
    public int VolonterId { get; set; }
    public int KorisnikId { get; set; }
    public string? KorisnikIme { get; set; }
    public string? KorisnikPrezime { get; set; }
    public string? KorisnikEmail { get; set; }
    public string? KorisnikTelefon { get; set; }
    public DateOnly DatumPridruzivanja { get; set; }
    public bool Aktivan { get; set; }
    public string? Napomena { get; set; }
    public decimal UkupnoSati { get; set; }
}

File: dogshelter_shared/lib/volonter/domain/volonter.dart
class Volonter {
  Volonter({
    required this.volonterId,
    required this.korisnikId,
    this.korisnikIme,
    this.korisnikPrezime,
    this.korisnikEmail,
    this.korisnikTelefon,
    required this.datumPridruzivanja,
    required this.aktivan,
    this.napomena,
    required this.ukupnoSati,
  });

  final int volonterId;
  final int korisnikId;
  final String? korisnikIme;
  final String? korisnikPrezime;
  final String? korisnikEmail;
  final String? korisnikTelefon;
  final DateTime datumPridruzivanja;
  final bool aktivan;
  final String? napomena;
  final double ukupnoSati;

  factory Volonter.fromJson(Map<String, dynamic> json) {
    return Volonter(
      volonterId: json['volonterId'] as int,
      korisnikId: json['korisnikId'] as int,
      korisnikIme: json['korisnikIme'] as String?,
      korisnikPrezime: json['korisnikPrezime'] as String?,
      korisnikEmail: json['korisnikEmail'] as String?,
      korisnikTelefon: json['korisnikTelefon'] as String?,
      datumPridruzivanja: DateTime.parse(json['datumPridruzivanja'] as String),
      aktivan: json['aktivan'] as bool,
      napomena: json['napomena'] as String?,
      ukupnoSati: (json['ukupnoSati'] as num).toDouble(),
    );
  }
}
